internal/brokers/examplebroker.go

```diff
@@ -99,10 +99,13 @@ func (b *exampleBroker) NewSession(ctx context.Context, username, lang string) (
 
 // GetAuthenticationModes returns the list of supported authentication modes for the selected broker depending on session info.
 func (b *exampleBroker) GetAuthenticationModes(ctx context.Context, sessionID string, supportedUILayouts []map[string]string) (authenticationModes []map[string]string, err error) {
-	session, ok := b.currentSessions[sessionID]
-	if !ok {
+	b.currentSessionsMu.RLock()
```

Comment:
It seems this code is repeated many time, do you mind creating a small private function here and always call it?

Reply:
Oh, thank you!! My hands were shaking to do it hahahaha